Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm

```diff
@@ -1011,6 +1012,14 @@ - (void)enterFullScreen:(CGSize)videoDimensions
 
         [CATransaction commit];
     }];
+
+#if ENABLE(QUICKLOOK_FULLSCREEN)
+        if (_isImageElement) {
+            id webViewUIDelegate = [webView UIDelegate];
+            if ([webViewUIDelegate respondsToSelector:@selector(_webView:didFullscreenImageWithQuickLook:)])
+                [webViewUIDelegate _webView:webView.get() didFullscreenImageWithQuickLook:mediaDimensions];
```

Comment:
I'm not sure this is the right place to call this method. There are many places that can still cancel fullscreen following this call.

I would recommend following an approach similar to the existing `_webViewDidEnterFullscreen:` SPI.

See `UIDelegate::UIClient::didEnterFullscreen` / `WebFullScreenManagerProxy::didEnterFullScreen` / `FullscreenClient::didEnterFullscreen`. 

Updating `FullscreenClient::didEnterFullscreen` is probably the most correct. You should be able to read state off of `WKFullScreenWindowController` via `[m_webView fullScreenWindowController]`.